### PR TITLE
Bookmarks: Add headphone controls settings screen

### PR DESCRIPTION
**Pull request by ashiagr**
## Description

This adds bookmarks headphone controls settings screen. 
## Testing Instructions
1. Install debug build
2. Go to `Profile` -> ⚙️  and tap the `Headphone controls` menu
3. ✅ Verify the view appears and looks good
4. ✅ Verify the default values are: 
    - `Previous action`: `Skip back`
    - `Next action`: `Skip forward`
5. Tap on the `Previous action` item
6. ✅ Verify the options are:
    - `Skip back`
    - `Previous chapter`
    - `Add bookmark`
9. Tap the `Previous chapter`  item
10. ✅ Verify the secondary label reflects the change
11. Tap back to close the view, then reopen it
12. ✅ Verify the setting is persisted
13. Tap the `Next action` item
14. ✅ Verify the options are:
    - `Skip forward`
    - `Next chapter`
    - `Add bookmark`
15. Tap the `Next chapter` option
16. ✅ Verify the label changes, and the value is persisted
14. Tap either option and select `Add bookmark`
15. ✅ Verify the `Bookmark confirmation sound` row appears
17. ✅ Verify the value is persisted again
18. Tap the same action, and select anything but A`dd bookmark`
19. ✅ Verify the `Bookmark confirmation sound` row is hidden
20. Repeat the steps with the other section
21. With the `Bookmark confirmation sound` row visible, tap the Switch and ✅ verify the value is persisted
22. Tap the `Bookmark confirmation sound` row and ✅ verify the value is toggled

Disabled feature flag

1. Go to `Profile` -> ⚙️ -> `Beta features` and toggle off the `Bookmarks` feature
2. Restart the app 
3. Go to `Profile` -> ⚙️ 
4. ✅ Notice that `Headphone controls` menu is hidden

## Screenshots or Screencast 

out

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

File: modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/HeadphoneControlsSettingsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                    }
+                )
+                if (shouldShowConfirmationSoundRow()) {
+                    ConfirmationSoundRow(
+                        saved = settings.headphonePlayBookmarkConfirmationSoundFlow.collectAsState().value,
+                        onSave = {
+                            settings.setHeadphoneControlsPlayBookmarkConfirmationSound(it)
+                        }
+                    )
+                }
```
Comment (geekygecko) [inline modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/HeadphoneControlsSettingsFragment.kt:98]: Would it be good to move the state up so it's easier to add a preview to the component?

```suggestion
                val previousAction = settings.headphonePreviousActionFlow.collectAsState().value
                val nextAction = settings.headphoneNextActionFlow.collectAsState().value
                val confirmationSound = settings.headphonePlayBookmarkConfirmationSoundFlow.collectAsState().value

                HeadphoneControlsSettingsPage(
                    previousAction = previousAction,
                    nextAction = nextAction,
                    onPreviousActionSave = { settings.setHeadphoneControlsPreviousAction(it) },
                    onNextActionSave = { settings.setHeadphoneControlsNextAction(it) },
                    confirmationSound = confirmationSound,
                    onConfirmationSoundSave = { settings.setHeadphoneControlsPlayBookmarkConfirmationSound(it) },
                    onBackPressed = {
                        @Suppress("DEPRECATION")
                        activity?.onBackPressed()
                    }
                )
            }
        }
    }

    @Composable
    private fun HeadphoneControlsSettingsPage(
        previousAction: HeadphoneAction,
        nextAction: HeadphoneAction,
        onPreviousActionSave: (HeadphoneAction) -> Unit,
        onNextActionSave: (HeadphoneAction) -> Unit,
        confirmationSound: Boolean,
        onConfirmationSoundSave: (Boolean) -> Unit,
        onBackPressed: () -> Unit,
    ) {
        Column {
            ThemedTopAppBar(
                title = stringResource(LR.string.settings_title_headphone_controls),
                bottomShadow = true,
                onNavigationClick = { onBackPressed() }
            )
            Column(
                modifier = Modifier
                    .verticalScroll(rememberScrollState())
            ) {
                TextP50(
                    text = stringResource(LR.string.settings_headphone_controls_summary),
                    color = MaterialTheme.theme.colors.primaryText02,
                    modifier = Modifier.padding(16.dp)
                )
                PreviousActionRow(
                    saved = previousAction,
                    onSave = onPreviousActionSave
                )
                NextActionRow(
                    saved = nextAction,
                    onSave = onNextActionSave
                )
                if (previousAction == HeadphoneAction.ADD_BOOKMARK || nextAction == HeadphoneAction.ADD_BOOKMARK) {
                    ConfirmationSoundRow(
                        saved = confirmationSound,
                        onSave = onConfirmationSoundSave
                    )
                }
```

File: modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/SettingsFragmentPage.kt [hunk 2]
```diff
@@ -229,6 +232,15 @@ private fun AutoAddToUpNextRow(onClick: () -> Unit) {
     )
 }
 
+@Composable
+private fun HeadphoneControlsRow(onClick: () -> Unit) {
+    SettingRow(
+        primaryText = stringResource(LR.string.settings_title_headphone_controls),
+        icon = GradientIconData(IR.drawable.ic_headphone),
```
Comment (geekygecko) [inline modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/SettingsFragmentPage.kt:239]: Should this be a tinted icon as we aren't using a gradient?

File: modules/services/images/src/main/res/drawable/ic_headphone.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="22dp"
+    android:height="19dp"
```
Comment (geekygecko) [inline modules/services/images/src/main/res/drawable/ic_headphone.xml:3]: Can we export this as 24dp by 24dp to match the other icons on the settings page?

**Review by geekygecko — APPROVED** [review]
That looks great. I only had a few comments.

File: modules/services/images/src/main/res/drawable/ic_headphone.xml (hunk 3 above)
Comment (ashiagr) [reply modules/services/images/src/main/res/drawable/ic_headphone.xml]: Updated in d1b9cf9

File: modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/HeadphoneControlsSettingsFragment.kt (hunk 1 above)
Comment (ashiagr) [reply modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/HeadphoneControlsSettingsFragment.kt]: I applied the changes and also added a preview in [a6b42ff](https://github.com/Automattic/pocket-casts-android/pull/1156/commits/a6b42ff993b26db56266af955645fcc672fc9736). Thanks for the suggestion. 🙏

File: modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/SettingsFragmentPage.kt (hunk 2 above)
Comment (ashiagr) [reply modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/SettingsFragmentPage.kt:239]: `GradientIcon` handles empty colors list internally so ignoring this one.

**Pull request merged**